### ISSUE TITLE
bcftools mpileup/call for variant calling from nanopore long read sequences

### ISSUE THREAD

**Issue by herrroaa** [samtools/bcftools#873]
Hi,

I have sequenced a single gene using naopore minion sequencer, aligned the reference genome, and I now want to call variants. I have some SNP genotypes for my target gene which I am going to use as a control for my sequence-based variant calls.
I wonder if there is a modified bcftools mpileup/call pipeline that fits this long reads with higher error rate?

Thanks\

**Comment by pd3** [comment]
Unfortunately not, bcftools does not have a special support for long reads and I am not aware of such a modification by anyone.

**Comment by mp15** [comment]
It's probably worth pointing out that given that the error mode with the current long read technologies is mostly INDELs rather than base substitutions the majority of the special support will be in the mapper not the variant caller. That said @jkbonfield was working on a realigner, which I suspect might help with these platforms if given the correct probabilities but the first step is definitely using the right mapper.

**Comment by pd3** [comment]
Yes, That is right. But note that even with correct alignments, `mpileup` does not handle well reads which contain multiple indels. It can miss valid variants due to underestimating their quality, an artifact of consensus building in `bcf_call_gap_prep()`.

**Comment by herrroaa** [comment]
Thanks for the reply !
indeed as @pd3  I can see in my data that mpileup missed several variants that appear to be true one even by looking to the alignment data only.

**Comment by herrroaa** [comment]
If you allow me, I have another question.
I have a bam file includes a single gene sequence with huge coverage (> 5000X) from which I want to call snps. From my original bam file , I will generate some bam files with different coverage. Then I will call SNPs from each one of them, compare them with the previously known genotypes to see which coverage gives me the highest concordance, and what is the coverage threshold below which snp calling is not reliable. I am goons use `samtools view -s 0.1 / 0.2 / 0.3 `and so on  togenrate these bam files.

**Q1:** Which reads will be excluded, is it a random process? Do I have any power to select on what bases these reads will be excluded? as for example keep the reads with highest mapping quality or base quality scores?

**Q2:** Since the coverage is not uniform across my target region, can I downsample to a certain number of reads at a position rather than using percentage?


Thanks
